### PR TITLE
Add Register fragment

### PR DESCRIPTION
**Pull request by jbsv**
Add a fragment to register a new user using its passport N°

### CODE REVIEW

File: android/app/src/main/java/com/epfl/dedis/hbt/ui/register/RegisterViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import javax.inject.Inject
+
+@HiltViewModel
+class RegisterViewModel @Inject constructor(private val userRepository: UserRepository) :
+    ViewModel() {
+
+    private val _registerForm = MutableLiveData<RegisterFormState>()
+    val registerFormState: LiveData<RegisterFormState> = _registerForm
+
+    private val _registerResult = MutableLiveData<LoginResult>()
```
Comment (jbsv) [inline android/app/src/main/java/com/epfl/dedis/hbt/ui/register/RegisterViewModel.kt:21]: not sure if "hijacking" the LoginResult is the most clever approach here ...

File: android/app/src/main/java/com/epfl/dedis/hbt/ui/register/RegisterViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    private fun isUserNameValid(username: String): Boolean {
+        return username.length >= 2
+    }
+
+    // A placeholder PIN code validation check
+    private fun isPincodeValid(pincode: String): Boolean {
+        return pincode.length in 4..9
+    }
+
+    // A placeholder PIN code validation check
```
Comment (jbsv) [inline android/app/src/main/java/com/epfl/dedis/hbt/ui/register/RegisterViewModel.kt:58]: comment is wrong

File: android/app/src/main/java/com/epfl/dedis/hbt/utility/NfcReader.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
+package com.epfl.dedis.hbt.utility
+
+import android.content.Context
+import android.nfc.NfcAdapter
+import android.nfc.NfcAdapter.getDefaultAdapter
+import android.widget.Toast
+
+class NfcReader(activity: Context?) {
```
Comment (jbsv) [inline android/app/src/main/java/com/epfl/dedis/hbt/utility/NfcReader.kt:8]: NOT to be reviewed now

File: android/app/src/main/res/layout/fragment_register.xml [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  xmlns:tools="http://schemas.android.com/tools"
+  android:id="@+id/container"
+  android:layout_width="match_parent"
+  android:layout_height="match_parent"
+  android:paddingHorizontal="@dimen/fragment_horizontal_margin"
+  android:paddingVertical="@dimen/fragment_vertical_margin"
+  tools:context=".ui.register.RegisterFragment">
+
+  <EditText
+    android:id="@+id/register_username"
```
Comment (jbsv) [inline android/app/src/main/res/layout/fragment_register.xml:13]: naming convention? 
`registerUsername`

File: android/app/src/main/java/com/epfl/dedis/hbt/data/model/User.kt [hunk 5]
```diff
@@ -5,5 +5,6 @@ package com.epfl.dedis.hbt.data.model
  */
 data class User(
     val name: String,
-    val pincode: Int
+    val pincode: Int,
+    val passport: String
```
Comment (GabrielFleischer) [inline android/app/src/main/java/com/epfl/dedis/hbt/data/model/User.kt:9]: Should this be a String ?

**Review by GabrielFleischer — COMMENTED** [review]
It looks mostly good to me

File: android/app/src/main/java/com/epfl/dedis/hbt/data/model/User.kt (hunk 5 above)
Comment (jbsv) [reply android/app/src/main/java/com/epfl/dedis/hbt/data/model/User.kt:9]: Not needed yet

**Pull request merged**